sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -1032,6 +1108,43 @@ private static Cluster getCluster(
     return builder.build();
   }
 
+  private static Cluster getCluster(
+      ValueProvider<String> hosts,
+      ValueProvider<Integer> port,
+      ValueProvider<String> username,
+      ValueProvider<String> password,
+      String encryptedPassword,
+      PasswordDecrypter passwordDecrypter,
+      ValueProvider<String> localDc,
+      ValueProvider<String> consistencyLevel) {
+    Cluster.Builder builder =
+        Cluster.builder().addContactPoints(hosts.get().split(",")).withPort(port.get());
+
+    if (username != null) {
+      String passwordStr = null;
+      if (encryptedPassword != null && passwordDecrypter != null) {
```

Comment:
@echauchot What is the rational on this extra 'password' should not this be simply the normal password and the logic be decided only with passwordDecrypter (if passwordDecrypter well we decrypt the password? Feels really weird to have two methods to pass the password.